Vis loading indication

w33ble opened the pull request:
Closes #2565

Add opacity change to visualizations when the search source has outstanding requests

lukasolson commented:
So, I'm not sure how involved this would be, but I like the way that discover puts an overlay spinner directly over the visualization in addition to changing opacity:

![spinner](https://cloud.githubusercontent.com/assets/1178348/6989580/2397cfde-da12-11e4-8705-574d27cfecc5.gif)

It might be overload on a dashboard to have a bunch of different spinners, but it's probably worth investigating.

Also, the doc table doesn't seem to be grayed out on the dashboard, while other visualization types do seem to.

w33ble commented:
Added the spinner to the visualize directive. I don't see the same issue with the data table though.

![2015-04-06 17_33_45](https://cloud.githubusercontent.com/assets/404731/7015088/ea4de588-dc82-11e4-8f49-ad3965aa8451.gif)

![2015-04-06 17_32_37](https://cloud.githubusercontent.com/assets/404731/7015090/ef44b85a-dc82-11e4-9e01-759960629eda.gif)

lukasolson commented:
It's not the data table that was having the problem... It's the "doc table" (i.e. saved search):

![saved-search](https://cloud.githubusercontent.com/assets/1178348/7026441/84e13a16-dcfe-11e4-872c-c8157644469a.gif)

lukasolson commented:
Also, I really like the spinner animation. Looks great!

w33ble commented:
Added loader to the doc-table. Also found a redundant spinner in discover, so I removed that as well.

![2015-04-07 13_32_11](https://cloud.githubusercontent.com/assets/404731/7033038/02d44a94-dd2c-11e4-9d82-255d5abe0efc.gif)

lukasolson commented:
So.... very minor thing here, but I don't think it makes sense to show the spinner on the markdown vis type, because it's not tied to a search response. Maybe you could check `vis.type.requiresSearch` before showing the spinner/overlay... :moneybag::moneybag:

w33ble commented:
@lukasolson good call. done and done :seedling:

lukasolson commented:
LGTM!
